x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportPutFollowAction.java

```diff
@@ -132,39 +142,49 @@ private void createFollowerIndex(
             return;
         }
 
-        final Settings.Builder settingsBuilder = Settings.builder()
-            .put(IndexMetaData.SETTING_INDEX_PROVIDED_NAME, request.getFollowRequest().getFollowerIndex())
-            .put(CcrSettings.CCR_FOLLOWING_INDEX_SETTING.getKey(), true);
-        final String leaderClusterRepoName = CcrRepository.NAME_PREFIX + request.getRemoteCluster();
-        final RestoreSnapshotRequest restoreRequest = new RestoreSnapshotRequest(leaderClusterRepoName, CcrRepository.LATEST)
-            .indices(request.getLeaderIndex()).indicesOptions(request.indicesOptions()).renamePattern("^(.*)$")
-            .renameReplacement(request.getFollowRequest().getFollowerIndex()).masterNodeTimeout(request.masterNodeTimeout())
-            .indexSettings(settingsBuilder);
+        boolean pre67CompatibilityMode = localClusterMinNodeVersion.before(Version.V_6_7_0)
+            || metaDataTuple.v1().getNodes().getMinNodeVersion().before(Version.V_6_7_0);
 
-        final Client clientWithHeaders = CcrLicenseChecker.wrapClient(this.client, threadPool.getThreadContext().getHeaders());
-        threadPool.executor(ThreadPool.Names.SNAPSHOT).execute(new AbstractRunnable() {
+        if (pre67CompatibilityMode) {
+            logger.warn("Pre-6.7 nodes present in local/remote cluster. Cannot bootstrap from remote. Creating empty follower index [{}] " +
```

Comment:
why log a warning? Isn't this just a normal thing? Log at debug level here.

Reply:
I think the same applies to any feature we introduce in a new version. If you are in a mixed version cluster, the feature will not work. The special case here is that a remote cluster might be on an older version, not supporting that feature. I think this is a natural extension of said logic. Just as for CCS, if you search over multiple cluster with a feature introduced in a newer version, that will not work in the presence of older clusters.